### ISSUE TITLE
Bad behavior generating new zettel when there is a dot in the name (double heading and backlinks)

### ISSUE THREAD

**Issue by IIPatternII** [michal-h21/vim-zettel#51]
While selecting a text and creating a new zettel, this happened:

![image](https://user-images.githubusercontent.com/48034748/88402039-efcef280-cda0-11ea-9ae6-68d056f761e3.png)

### I manage to isolate the problem (the dot). To reproduce follow these steps:

1. Write any text inside a zettel whit a dot somewhere, for example:
**A Concise Introduction to Logic - Patrick J. Hurley**
![image](https://user-images.githubusercontent.com/48034748/88402496-91564400-cda1-11ea-9a22-3b4ce791869c.png)
2. "V" to select the entire line, and "z" to create a new zettel:
![image](https://user-images.githubusercontent.com/48034748/88402650-cd89a480-cda1-11ea-82a8-326669d06fb4.png)
3. Repeat step 2 (inside the generated zettel) but with any text that you want and this will show up:
![image](https://user-images.githubusercontent.com/48034748/88402857-15a8c700-cda2-11ea-8d7a-ca4f02c7a3cc.png)
4. hit enter and this is the zettel generated:
![image](https://user-images.githubusercontent.com/48034748/88402943-3244ff00-cda2-11ea-8cac-4d8708c940b1.png)

**It does not matter what the title says as long as there is a dot somewhere.**

**Comment by michal-h21** [comment]
Thanks for the report. I cannot reproduce it, unfortunately. Could you please share also your configuration for Vim-zettel?

**Comment by IIPatternII** [comment]
Thanks to you for the proyect! Sure!

OS: MacOS 10.15.4
Vim: 8.2.539

This is my vimrc:
```
" Para que funcione bien vimwiki
set nocompatible              " be iMproved, required
filetype off                  " required
syntax on

" set the runtime path to include Vundle and initialize
set rtp+=~/.vim/bundle/Vundle.vim
call vundle#begin()
" let Vundle manage Vundle, required
Plugin 'VundleVim/Vundle.vim'

"FuzzyFider
Plugin 'junegunn/fzf'
Plugin 'junegunn/fzf.vim'

Plugin 'tpope/vim-fugitive'
Plugin 'git://git.wincent.com/command-t.git'
Plugin 'rstacruz/sparkup', {'rtp': 'vim/'}

" Zettelkasten
Plugin 'vimwiki/vimwiki'
Plugin 'michal-h21/vim-zettel'

"Tema
Plugin 'morhetz/gruvbox'

"Soncronizar automaticamente Github
Plugin 'michal-h21/vimwiki-sync'

call vundle#end()            " required
filetype plugin indent on    " required

call plug#begin('~/.vim/plugged')
Plug 'https://github.com/alok/notational-fzf-vim'
call plug#end()

"----------------------------- VimWiki Zettelkasten ----------------------------------------
" Settings for Vimwiki
let g:vimwiki_list = [{'path':'~/scratchbox/vimwiki/zettelkasten/','ext':'.md','syntax':'markdown'}]
let g:vimwiki_autowriteall = 1
"let g:zettel_fzf_command = 
let g:zettel_format = "%y-%m-%d_%H%M_%title"
let g:zettel_options = [{"template" :  "~/scratchbox/vimwiki/template1.tpl"}]
let g:nv_search_paths = ['~/scratchbox/vimwiki/zettelkasten']

"nnoremap <leader>d :ZettelBackLinks<CR>
nnoremap <C-p> :ZettelOpen<CR>

"-----------------------------  Visual  ----------------------------------------
set relativenumber
set tabstop=4 softtabstop=4
set nowrap
set smartcase
set termguicolors

set colorcolumn=100
highlight ColorColumn ctermbg=0 guibg=lightgrey

autocmd vimenter * colorscheme gruvbox

"-----------------------------  Splits  ----------------------------------------
let mapleader = " "
set timeout timeoutlen=3000 ttimeoutlen=100
set splitright
set splitbelow

nnoremap <leader>h :wincmd h<CR>
nnoremap <leader>j :wincmd j<CR>
nnoremap <leader>k :wincmd k<CR>
nnoremap <leader>l :wincmd l<CR>
nnoremap <Leader>+ :vertical resize +5<CR>
nnoremap <Leader>- :vertical resize -5<CR>

"-----------------------------  Escribir  ----------------------------------------

nnoremap ; :
nnoremap : ; 
inoremap ; :
inoremap : ; 

"-----------------------------  Usabilidad  ----------------------------------------
set noerrorbells
nnoremap <C-s> :w<CR>
" Convertir palabr en tag
nnoremap <leader>t bi:<ESC>t<Space>a:<ESC> 
```
Thank you again!

**Comment by michal-h21** [comment]
I still cannot reproduce it, I am afraid :/  But I can see that you have mappings for `:` in the `vimrc` file. I've found that I use `execute ":w"` on the quite a few places in `vim-zettel`, so maybe this is the problem. I found that it is not necessary to use `:w`, `w` is enough, so I replaced them. Does this fix it for you?

**Comment by IIPatternII** [comment]
Thank you for the response. I tried so many things by now... and what I have to prove it you ask? A new bug... haha. I removed everything from the **.vimrc** except for the plugins and the minimal vim-zettel and Vimwiki config.

Now the previous bug disappeared, but there is a new one. When I create a note inside another note, the link is generated, but not the new note. So, when I hit **Enter** on the link, it goes nowhere. 

**By the way, the thing that made the previous bug disappear was commenting this line:**
`let g:zettel_format = "%y-%m-%d_%H%M_%title"`
I can't say why.

I will reinstall everything and report back. Luckily my work setup is safe on Neovim.
